cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -389,6 +394,9 @@ class reader::impl {
   std::vector<std::unique_ptr<datasource>> _sources;
   std::unique_ptr<aggregate_reader_metadata> _metadata;
 
+  // Number of sources
+  size_t _num_sources{0};
```

Comment:
Needed as `_sources` is not available in the experimental reader subclass so the `_sources.size()` which was being used in some member functions